packages/oui-datagrid/src/datagrid.controller.js

```diff
@@ -246,25 +246,18 @@ export default class DatagridController {
         this.refreshData(() => {
             this.paging.setOffset(1);
             this.paging.setCriteria(criteria);
-        }, false, false);
+        }, false);
     }
 
-    onPaginationChange ($event) {
+    onPaginationChange ({ offset, pageSize }) {
         this.refreshData(() => {
-            this.paging.setOffset($event.offset);
-            this.paging.setPageSize($event.pageSize);
-        }, true, true);
+            this.paging.setOffset(offset);
+            this.paging.setPageSize(pageSize);
+            this.$scope.$emit("oui-datagrid.onPaginationChange", { offset, pageSize });
```

Comment:
Do you use this in your code ? I don't see a `$on('oui-datagrid.onPaginationChange')` here.
Why not callbacks `onPaginationChange` and `onDataRefresh` ?

Reply:
I guess this is for users, it can be used outside the Component

Reply:
Indeed, this allows users to have custom behaviour on pagination change if they want (like using `scrollIntoView` if they want to :)

Reply:
This doesn't answer my last question above :)

Why prefer using `$emit` instead of callbacks ?
We could achieve exactly the same thing with callbacks too.

For me, it's more web-components friendly callbacks than emit/broadcast.

Reply:
Wouldn't a callback for each event be worse than events ?

Reply:
@FredericEspiau are you talking in terms of performances ? 
It's true that for consistency a callback would be better